src/relay/op/tensor/transform.h

```diff
@@ -180,6 +181,134 @@ static inline Array<Array<Layout>> ConcatenateLayout(const Attrs& attrs,
   return Array<Array<Layout>>{Array<Layout>(old_in_layouts.size(), ret), {ret}};
 }
 
+static inline Array<IndexExpr> infer_newshape(const Array<IndexExpr>& data_shape,
```

Comment:
I suggest we just declare the function in the header and define it in the .cc file. Inlining such a large function is usually not preferable though it is the compiler's decision to inline it or not.